### DISCUSSION TITLE
Popup onClick Without React

### DISCUSSION THREAD

**Discussion by trk17** [visgl/deck.gl#7943] [Q&A]
Hello!

I am using Deck without React in an Angular application and looking to display a popup when an Icon in an IconLayer is clicked. The popup would then move/pane/zoom alongside the map. Basically the example shown here: https://docs.mapbox.com/mapbox-gl-js/example/popup-on-click/

I have found a few examples of this using react (https://stackoverflow.com/questions/46310202/deck-gl-how-to-show-popup-onclick) but have been unable to find any without.

Capturing the onClick and rendering a custom component is quite trivial but keeping that component pointed at the Icon is proving much more difficult. I could listen to events from the ViewState to move the components absolute position, but that seems quite inefficient and error-prone.

Appreciate any insight here. Thanks

**Comment by lokendra9719** [comment]
Hi, Team any update on this ? facing the same problem don't want to use mapbox-gl

**Comment by chrisgervang** [comment]
deck.gl shipped an [InfoWidget](https://deck.gl/docs/api-reference/widgets/info-widget) with a click mode in [9.3](https://deck.gl/docs/whats-new#deckgl-v93) that is designed to make this trivial to add and customize. The widget system is framework agnostic and can be [themed](https://deck.gl/docs/api-reference/widgets/styling) with CSS variables. Please let us know if it works for your use case!
